src/object_store/cache.rs

```diff
@@ -41,20 +40,14 @@ impl CacheFileManager {
         Self { base_path }
     }
 
-    async fn write_file(
-        &self,
-        cache_key: &CacheKey,
-        data: Arc<Bytes>,
-    ) -> io::Result<PathBuf> {
+    async fn write_file(&self, cache_key: &CacheKey, data: Bytes) -> io::Result<PathBuf> {
         let mut path = self.base_path.to_path_buf();
         path.push(cache_key.as_filename());
 
-        // Should this happen normally?
+        // TODO: when does this happen?
```

Comment:
I guess when:

thread 1 evicts the item from cache, notifies the eviction listener (but that doesn't delete the file yet)
thread 2 tries to get the item, cache miss, downloads it, tries to write to the file

I wish there were some kind of moka semantics where the cache entry stays there as long as and only as long as the eviction listener hasn't finished :D

Reply:
Yeah, agreed with both comments.